packages/framework/src/Model/Product/View/ListedProductView.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Model\Product\View;
+
+use Shopsys\FrameworkBundle\Model\Product\Pricing\ProductPrice;
+use Shopsys\FrameworkBundle\Component\Image\View\ImageView;
+
+/**
+ * @experimental
+ *
+ * Class representing products in lists in FE templates (to avoid usage of Doctrine entities a hence achieve performance gain)
+ */
+class ListedProductView
+{
+    /**
+     * @var int
+     */
+    protected $id;
+
+    /**
+     * @var string
+     */
+    protected $name;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\Image\View\ImageView|null
+     */
+    protected $image;
+
+    /**
+     * @var string
+     */
+    protected $availability;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Product\Pricing\ProductPrice
+     */
+    protected $sellingPrice;
+
+    /**
+     * @var ?string
```

Comment:
according to https://docs.phpdoc.org/guides/types.html it should be `string|null`